### ISSUE TITLE
Syncthing running on detached tmux won't restart after automatic update

### ISSUE THREAD

**Issue by hpello** [syncthing/syncthing#2144]
Hello,

On Ubuntu 14.10, I recently tried and have syncthing run on a detached tmux session:
```
tmux new-session -d -s syncthing ./syncthing
```

It all goes fine, but after a weekly automatic update, I usually find out that syncthing has stopped running. The logs say:
```
[XXXXX] 10:49:16 INFO: Automatic upgrade (current "v0.11.18" < latest "v0.11.19")
[XXXXX] 10:49:20 WARNING: Automatically upgraded to version "v0.11.19". Restarting in 1 minute.
[XXXXX] 10:50:20 OK: Exiting
[monitor] 10:50:20 INFO: Restarting monitor...
```
... and that's all.

However, when syncthing is running on the detached tmux session, and I choose to restart it manually from the GUI, the restart works without any problem:
```
[XXXXX] 08:09:40 INFO: Restarting
[XXXXX] 08:09:40 OK: Exiting
[monitor] 08:09:40 INFO: Syncthing exited: exit status 3
[monitor] 08:09:41 INFO: Starting syncthing
[XXXXX] 08:09:43 INFO: syncthing v0.11.19 (go1.4.2 linux-amd64 default) unknown-user@syncthing-builder 2015-08-09 09:56:22 UTC
```

Is this normal behaviour? 

Thank you

**Comment by wweich** [comment]
This is "kind of" normal behavior.
When you restart manually you are telling the syncthing monitor process to restart the syncthing instance.
When an updates is performed, the monitor itself has to be restarted.
I think that syncthing would then be started outside the tmux session. Did you check your processes?

**Comment by hpello** [comment]
@wweich Oh I see. These are two different kinds of update then.

Well, I switched back to the previous version (`cp syncthing.old syncthing`) and ran syncthing again the tmux session.

Everything goes well:
```
> ps aux | grep "[s]yncthing"
hippo    20744  0.2  0.6 192044  7036 pts/5    Ssl+ 08:49   0:00 /home/hippo/libs/syncthing
hippo    20749  0.0  0.0   4464   664 ?        S    08:49   0:00 sh -c cat >> /home/hippo/logs/syncthing.log
hippo    20752  1.9  0.4   9332  4676 pts/9    Ssl+ 08:49   0:00 /home/hippo/libs/syncthing-inotify
hippo    20760 45.0  3.3 435588 33888 pts/5    Sl+  08:49   0:15 /home/hippo/libs/syncthing
hippo    20772  0.0  0.0   7256   632 pts/4    S+   08:49   0:00 tail -F logs/syncthing.log
```

The automatic update is triggered:
```
[XXXXX] 08:49:14 INFO: Automatic upgrade (current "v0.11.18" < latest "v0.11.19")
[XXXXX] 08:49:19 WARNING: Automatically upgraded to version "v0.11.19". Restarting in 1 minute.
[XXXXX] 08:50:19 OK: Exiting
[monitor] 08:50:19 INFO: Restarting monitor...
```

However after the automatic update, no track of the syncthing process:
```
> ps aux | grep "[s]yncthing"
hippo    20752  0.9  0.4   9332  4676 pts/9    Ssl+ 08:49   0:00 /home/hippo/libs/syncthing-inotify
hippo    20772  0.0  0.0   7256   632 pts/4    S+   08:49   0:00 tail -F logs/syncthing.log
```

**Comment by AudriusButkevicius** [comment]
The monitor invokes itself, detaches, and exits, so tmux is somehow interfering with that.

**Comment by xJMV** [comment]
It is the same in Windows... shut it down and restart it. This is annoying for a remote machine... We just need to remember that Syncthing is in dev so it won't be perfect for a while.

I have no idea if it is possible but OLD monitor should launch NEW monitor and watch if it gets alive properly prior to shutdown itself. If NEW doesn't start properly, OLD shut down NEW and shows an alert to have Syncthing restarted manually by the user.

**Comment by hpello** [comment]
Hi,

Thanks to dbb3887, that fixed #2169, after an upgrade the monitor now restarts with the same PID (on UNIX systems).

I tested the automatic upgrade from 0.11.21 to 0.11.22 on a linux machine. Syncthing was running on a detached tmux session and it survived the upgrade. As a consequence I think this issue is resolved.

**Comment by calmh** [comment]
:+1:
